# BHA0006 - Wrong type

Checks whether the type is correct.

````csharp
public class Model
{
    public int Field;
}

private static readonly AccessTools.FieldRef<object, string>? Field =
  AccessTools2.FieldRefAccess<object>("Custom.Namespace.Model:Field");

// Should be
private static readonly AccessTools.FieldRef<object, int>? Field =
  AccessTools2.FieldRefAccess<object>("Custom.Namespace.Model:Field");
````
